python_benchmarks/qr_codes/qr.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python
+
+"""qr.py
+Usage example:
+python qr.py -o out.yaml -p qrcodes/detection
+-H, --help - show help
+-o, --output - output file (default out.yaml)
+-p, --path - input dataset path (default qrcodes/detection)
+-a, --accuracy - input accuracy (default 47)
+-alg, --algorithm - input alg (default opencv)
+-m, --metric - input norm (default l_inf)
+"""
+
+import argparse
+import glob
+from enum import Enum
+
+import numpy as np
+import cv2 as cv
+
+
+class DetectorQR:
+    TypeDetector = Enum('TypeDetector', 'opencv opencv_wechat')
+
+    def __init__(self, type_detector=TypeDetector.opencv, path_to_model="./"):
```

Comment:
It looks like `type_detector` is redundant and used in factory method only. I propose to drop it here and in inherited classes.

Reply:
type_detector was removed from classes